duplex_http_call.go

```diff
@@ -276,3 +275,17 @@ func (d *duplexHTTPCall) getError() error {
 	defer d.errMu.Unlock()
 	return d.err
 }
+
+// See: https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/net/http/clone.go;l=22-33
+func cloneURL(u *url.URL) *url.URL {
```

Comment:
This was trivial enough to grab from stdlib and it's not exposed publicly. Was actually surprised there was no `(*url.URL).Clone()` or similar.